### PR TITLE
bug fix and feature improve of mongo writer, iterator and describe stream

### PR DESCRIPTION
**Pull request by lzc1104**
1.Fix describe stream reach upper limit
2.Add implementation of closed of mongo writer to recycle idle connection using delay strategy to prevent batcher later write
3.Use trim horizon strategy for split shard which prepares to get records, due to the inaccuracy of describe stream output which related to  potential skips part of shard data issue

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,28 @@
+2021-05-18
```
Comment (vinllen) [inline ChangeLog:1]: 这里可以加一下最新提交的change log，比如v1.0.12版本

File: conf/nimo-shake.conf [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # current configuration version, do not modify.
 # 当前配置文件的版本号，请不要修改该值。
-conf.version = 1
+conf.version = 3
```
Comment (vinllen) [inline conf/nimo-shake.conf:3]: 整体代码现在diff率太高了，看看能不能rebase到最新的版本，然后再进行合入，这样code review方便看一些

**Comment by vinllen** [comment]
另外，分支的话，合入到feature-1.0分支吧，master分支合入比较晚

**Review by zhangst — APPROVED** [review]

**Comment by zhangst** [comment]
Lots of code conflicts, merged code by hand. 
Thanks

**Pull request closed without merging**